common/functions/src/scalars/function_factory.rs

```diff
@@ -37,13 +38,21 @@ use crate::scalars::UdfFunction;
 
 pub type FactoryCreator = Box<dyn Fn(&str) -> Result<Box<dyn Function>> + Send + Sync>;
 
+// Temporary adaptation for arithmetic.
+pub type ArithmeticCreator =
+    Box<dyn Fn(&str, Vec<DataField>) -> Result<Box<dyn Function>> + Send + Sync>;
+
 #[derive(Clone)]
 pub struct FunctionFeatures {
     pub is_deterministic: bool,
     pub negative_function_name: Option<String>,
     pub is_bool_func: bool,
     pub is_context_func: bool,
     pub maybe_monotonic: bool,
+    pub num_arguments: usize,
+    // (1, 2) means we only accept [1, 2] arguments
+    // None means it's not variadic function
+    pub variadic_arguments: Option<(usize, usize)>,
```

Comment:
how about unify `num_arguments` and `variadic_arguments` into a single (usize, usize) in this struct?

calling num_arguments(2) -> set as (2, 2)
calling variadic_arguments(1, 2) -> set as (1, 2)

Reply:
> how about unify `num_arguments` and `variadic_arguments` into a single (usize, usize) in this struct?
> 
> calling num_arguments(2) -> set as (2, 2) calling variadic_arguments(1, 2) -> set as (1, 2)

A good idea.

Reply:
> > how about unify `num_arguments` and `variadic_arguments` into a single (usize, usize) in this struct?
> > calling num_arguments(2) -> set as (2, 2) calling variadic_arguments(1, 2) -> set as (1, 2)
> 
> A good idea.

If the changes big in this PR, I suggest we do it in another PR later, create a new issue first to track it.

Reply:
> variadic_arguments

Sure, I will create a new issue and PR.

Reply:
I prefer to keep as origin, it's strange to set `(2,2)` for none variadic argument function but most of the functions are none-variadic .